torchvision/datasets/vision.py

```diff
@@ -29,7 +29,7 @@ class VisionDataset(data.Dataset):
 
     def __init__(
         self,
-        root: Optional[str] = None,
+        root: str = None,  # type: ignore[assignment]
```

Comment:
can we still keep the `Optional[str]` type which is technically correct?

Reply:
We can't. If we use `Optional[str]` as annotation, `mypy` correctly infers that for the `self.root` attribute. And with that we are back at the issues we are currently seeing, i.e. all child datasets that perform anything with `self.root` would now need to check that `self.root` is in fact a `str` and not `None`.